package.json

```diff
@@ -30,8 +30,10 @@
     "eslint-config-import": "^0.13.0",
     "eslint-config-scratch": "^5.0.0",
     "eslint-plugin-import": "^2.8.0",
+    "jimp": "0.2.27",
```

Comment:
Cool! Jimp looks really useful!

I'm curious how fast it is. Do you mind logging it to see how long the resize takes?

Reply:
It is slow for large images, I'll try logging the numbers and share results -- but I think speed varies based on which resizing algorithm we choose. From the [wiki description here](https://en.wikipedia.org/wiki/Image_scaling), nearest neighbors sounded like the closest to what we were talking about (4 pixels per pixel), but we have other algorithms to choose from depending on what we want.

Reply:
here's my super scientific benchmark...
![image](https://user-images.githubusercontent.com/1786240/39136473-5064c980-46e9-11e8-8931-e345053cdaab.png)

Reply:
These numbers are from GUI -- right before importBitmap is called and right after we get the resized data back in our callback

Reply:
Hmm, it's kind of slow... we might want to think about putting a loading bar/spinner in there somewhere

Reply:
> nearest neighbors sounded like the closest to what we were talking about (4 pixels per pixel), but we have other algorithms to choose from depending on what we want.

I think nearest neighbor is the way to go - it's the one that'll match Scratch 2.0 behavior.

> Hmm, it's kind of slow... we might want to think about putting a loading bar/spinner in there somewhere

I agree. I also wonder if it would be faster to load the image into an `<img>` tag and stamp it onto a canvas with scaling. That should only be a few lines of code, but it does mean that image scaling would depend on the browser and, in particular, on the browser's implementation of the canvas `imageSmoothingEnabled` property and/or the CSS `image-rendering` property. See here: https://stackoverflow.com/questions/18547042/resizing-a-canvas-image-without-blurring-it

There's also the other answer on that page, but I imagine that's roughly what Jimp is doing. That said, it might be worth doing something like that in order to remove Jimp as a dependency, since I imagine it's fairly large.